src/main/scala/com/teragrep/functions/dpf_03/RegexTokenizerUDF.java

```diff
@@ -0,0 +1,86 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+
+package com.teragrep.functions.dpf_03;
+
+import com.teragrep.blf_01.Token;
+import com.teragrep.blf_01.Tokenizer;
+import org.apache.spark.sql.api.java.UDF2;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RegexTokenizerUDF implements UDF2<String, String, List<byte[]>> {
+
+    private Tokenizer tokenizer = null;
+
+    @Override
+    public List<byte[]> call(String s, String regex) {
+        if (tokenizer == null) {
+            // "lazy" init
+            tokenizer = new Tokenizer(0);
+        }
+
+        // create empty Scala immutable List
+        ArrayList<byte[]> rvList = new ArrayList<>();
+
+        ByteArrayInputStream bais = new ByteArrayInputStream(s.getBytes(StandardCharsets.UTF_8));
+        List<Token> tokens = tokenizer.tokenize(bais);
+
+        for (Token token : tokens) {
+            if (token.toString().matches(regex)) {
+                rvList.add(token.bytes);
+            } else if (regex.isEmpty()) {
```

Comment:
Is it more performant to check isEmpty() first and then matches if it isn't empty? Or is the performance difference negligible either way

Reply:
combined the checks to `regex.isEmpty() || pattern.matcher(token.toString()).matches()`.